products/views.py

```diff
@@ -1,3 +1,68 @@
-from django.shortcuts import render
+from enum import Enum
 
-# Create your views here.
+from django.http      import JsonResponse
+from django.views     import View
+from django.db.models import Q, F, Max, Min, Count
+
+from products.models import Product
+
+class BidTypeEnum(Enum):
+    BUY      = 1
+    SELL     = 2
+    END      = 3
+    CANCELED = 4
+
+class ProductListView(View):
+    def get(self, request):
+        category          = request.GET.get('category')
+        search            = request.GET.get('search')
+        shoe_size_list    = request.GET.getlist('shoe_size')
+        apparel_size_list = request.GET.getlist('apparel_size')
+        price_filter      = request.GET.get('price')
+        sort              = request.GET.get('sort')
+
+        q = Q()
+        q &= Q(productsize__bids__type_id=BidTypeEnum.BUY.value) if sort == 'sell_now' \
+            else Q(productsize__bids__type_id=BidTypeEnum.SELL.value)
+
+        if category:
+            q &= Q(category__name=category)
+
+        if search:
+            q &= Q(Q(korean_name__contains=search) | Q(english_name__icontains=search))
+
+        if shoe_size_list:
+            q &= Q(sizes__name__in=shoe_size_list)
+
+        if apparel_size_list:
+            q &= Q(sizes__name__in=apparel_size_list)
+
+        if price_filter:
+            price_list = price_filter.split('-')
+            q &= Q(buy_price__gte=price_list[0], buy_price__lte=price_list[1])
```

Comment:
```suggestion
            q &= Q(buy_price__range=[price_list[0], price_list[1])
```

다음과 같이 줄여 볼 수 있을 것 같습니다.

Reply:
리뷰해주신대로 수정했습니다